rfd/0144-client-tools-updates.md

```diff
@@ -0,0 +1,320 @@
+user is on a low bandwidth connection (LTE or public Wi-Fi), if the Teleport
+download server is inaccessible, or the user urgently needs to access the
+cluster and can not wait for the update to occur.
+
+```
+$ tsh login --proxy=proxy.example.com
+Client tools are out of date, updating to vX.Y.Z.
+Update progress: [▒▒▒▒▒▒     ] (Ctrl-C to cancel update)
+
+[...]
+```
+
+After downloading, client tools will re-execute the downloaded version.  The
+child process will inherit all environment variables and flags.
+`TELEPORT_TOOLS_VERSION=off` will be added during re-execution to prevent
+infinite loops.
+
+An environment variable `TELEPORT_TOOLS_VERSION` will be introduced that can be
+`X.Y.Z` (use specific version) or `off` (do not try to update). This
+environment variable can be used as a emergency workaround for a known issue,
+pinning to a specific version in CI/CD, or for debugging.
+
+Automatic updates will not be used if `tctl` is connecting to the Auth Service
+over localhost.
+
+##### Errors and warnings
+
+If the user does not enroll in automatic updates and the cluster administrator
+does not self-manage client tools updates as outlined in [Self-managed client
+tools updates](#self-managed-client-tools-updates), a series of warnings and
+errors with increasing urgency will be shown to the user.
+
+If the version of client tools is within the same major version as advertised
+by the cluster, a warning will be shown to urge the user to enroll in automatic
+updates. Warnings will not prevent the user from using client tools that are
+slightly out of date.
+
+```
+$ tsh login --proxy=proxy.example.com
+Warning: Client tools are out of date, update to vX.Y.Z.
+
+Update Teleport to vX.Y.Z from https://goteleport.com/download or your system
+package manager.
+
+Run "tsh autoupdate enroll" to enroll in automatic updates and keep client
+tools like tsh and tctl automatically updated.
+
+[...]
+```
+
+If the version of client tools is 1 major version below the version advertised
+by the cluster, a warning will be shown that indicates some functionality may
+not work.
+
+```
+$ tsh login --proxy=proxy.example.com
+WARNING: Client tools are 1 major version out of date, update to vX.Y.Z.
+
+Some functionality may not work. Update Teleport to vX.Y.Z from
+https://goteleport.com/download or your system package manager.
+
+Run "tsh autoupdate enroll" to enroll in automatic updates and keep client
+tools like tsh and tctl automatically updated.
+```
+
+If the version of client tools is 2 (or more) versions lower than the version
+advertised by the cluster or 1 (or more) version greater than the version
+advertised by the cluster, an error will be shown and will require the user to
+use the `--skip-version-check` flag.
+
+```
+$ tsh login --proxy=proxy.example.com
+ERROR: Client tools are N major versions out of date, update to vX.Y.Z.
+
+Some functionality will not work. Update Teleport to vX.Y.Z from
```

Comment:
I'd use a strong message here and say something like "Your cluster requires tsh vN or later"